src/MathExpression.cpp

```diff
@@ -0,0 +1,181 @@
+#include "MathExpression.h"
+
+#include <cctype>
+#include <sstream>
+#include <set>
+#include <vector>
+
+#include "Result.h"
+
+
+
+MathExpression::MathExpression(std::string equation)
+: equation_(sanitizeEquation(equation)) {};
+
+
+
+std::set<std::string> MathExpression::getVariableNames() const
+{
+    std::set<std::string> variablesList;
+
+    std::string variableName = "";
+    for (int i = 0; i < equation_.length(); i++)
+    {
+        variableName = "";
+        if (isalpha(equation_[i]))
+        {
+            while (isalpha(equation_[i]) && i < equation_.length())
```

Comment:
You're supporting variables longer than a single character? Nice. I like this. When I did this, I just made variables single characters like x, y, z.